src/controllers/user.controller.js

```diff
@@ -25,31 +26,23 @@ export default class UserController {
         preferred_currency: req.body.preferred_currency,
         location: req.body.location
       });
-
-      const userToken = auth.generateToken(user);
-      return res.status(201).json({
-        status: 'success',
-        data: {
-          id: user.id,
-          first_name: user.first_name,
-          last_name: user.last_name,
-          email: user.email,
-          role: user.role,
-          gender: user.gender,
-          birth_date: user.birth_date,
-          preferred_language: user.preferred_language,
-          preferred_currency: user.preferred_currency,
-          location: user.location,
-          isVerified: user.isVerified,
-          token: userToken
-        }
-      });
+      user.dataValues.token = UserUtils.generateToken(user);
+      return Responses.getSuccessResponse(res,
+        UserUtils.getPublicUserFields(user.dataValues), 201);
     } catch (error) {
-      return res.status(500)
-        .json({
-          status: 'error',
-          error: 'Could not signup user',
-        });
+      return Responses.getInternalServerErrorResponse(res, 'Could not signup user');
     }
   }
+
+  /**
+   * Handles signin requests
+   * @param {ServerRequest} req
+   * @param {ServerResponse} res
+   * @returns {ServerResponse} response
+   */
+  static signin(req, res) {
```

Comment:
Your signin controller seems not to be making any interaction with the database. Could you please explain how the user signs in?

Reply:
Yes, but I'll change that. I was implementing `passport-local` as a middleware which requires the the db interaction in its configuration. I,  however, removed it (passport) but it's ways kind of stuck with me. I'll changes and implement the db interaction in the controller instead

Reply:
Oh I see. No problem.